src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4296,6 +4297,117 @@ def _check_zip_deployment_status(cmd, rg_name, name, deployment_status_url, slot
     return res_dict
 
 
+def _get_latest_deployment_id(cmd, rg_name, name, deployment_status_url, slot):
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+
+    headers = get_scm_site_headers(cmd.cli_ctx, name, rg_name, slot)
+    total_trials = 30
+    num_trials = 0
+    while num_trials < total_trials:
+        time.sleep(2)
+        response = requests.get(deployment_status_url, headers=headers,
+                                verify=not should_disable_connection_verify())
+        try:
+            res_dict = response.json()
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.warning("Deployment status endpoint %s returned malformed data. Exception: %s "
+                           "\nRetrying...", deployment_status_url, ex)
+            return None
+        finally:
+            num_trials = num_trials + 1
+        if 'id' in res_dict and 'temp' not in res_dict['id']:
+            return res_dict['id']
+    return None
+
+
+def _track_deployment_runtime_status(params, deploymentstatusapi_url, deployment_id, timeout=None):
+    max_time_sec = int(timeout) if timeout else 1000
+    start_time = time.time()
+    time_elapsed = 0
+    deployment_status = None
+    while time_elapsed < max_time_sec:
+        response_body = send_raw_request(params.cmd.cli_ctx, "GET", deploymentstatusapi_url).json()
+        deployment_properties = response_body.get('properties')
+        deployment_status = deployment_properties.get('status')
+        time_elapsed = int(time.time() - start_time)
+        status = RUNTIME_STATUS_TEXT_MAP.get(deployment_status)
+        status = deployment_status if status is None else status
+        logger.warning("Status: %s Time: %s(s)", status, time_elapsed)
+        if deployment_status == "RuntimeStarting":
+            logger.info("InprogressInstances: %s, SuccessfulInstances: %s",
+                        deployment_properties.get('numberOfInstancesInProgress'),
+                        deployment_properties.get('numberOfInstancesSuccessful'))
+        if deployment_status == "RuntimeSuccessful":
+            break
+        if deployment_status == "RuntimeFailed":
+            error_text = ""
+            if int(deployment_properties.get('numberOfInstancesSuccessful')) > 0:
+                total_num_instances = int(deployment_properties.get('numberOfInstancesInProgress')) + \
+                    int(deployment_properties.get('numberOfInstancesSuccessful')) + \
+                    int(deployment_properties.get('numberOfInstancesFailed'))
+                error_text += "Site started with errors: {}/{} instances failed to start successfully\n".format(
+                    deployment_properties.get('numberOfInstancesFailed'),
+                    total_num_instances)
+            else:
+                error_text += "Deployment failed because the site failed to start within timeout limits.\n"
```

Comment:
`start within timeout limits.` -> can we put the actual timeout limit used here?